### PR TITLE
Hack around Log4j bug rendering exceptions

### PR DESCRIPTION
**Pull request by jasontedor**
Log4j has a bug where it does not handle a security exception that can
be thrown when it is rendering a stack trace. This commit intentionally
introduces jar hell with the ThrowableProxy class to work around this
bug until a fix is a released.

Relates #20304

### CODE REVIEW

**Comment by abeyad** [comment]
LGTM

File: core/src/main/java/org/elasticsearch/bootstrap/JarHell.java [hunk 1]
```diff
@@ -277,6 +277,9 @@ static void checkClass(Map<String,Path> clazzes, String clazz, Path jarpath) {
                 if (clazz.equals("org.joda.time.base.BaseDateTime")) {
                     return; // apparently this is intentional... clean this up
                 }
+                if (clazz.startsWith("org.apache.logging.log4j.core.impl.ThrowableProxy")) {
+                    return; // deliberate to hack around a bug in Log4j
```
Comment (s1monw) [inline core/src/main/java/org/elasticsearch/bootstrap/JarHell.java]: can we have a reference to the actual issue in here. It's just easier to reason about this with a direct link?

**Comment by s1monw** [comment]
left one comment LGTM otherwise

File: core/src/main/java/org/elasticsearch/bootstrap/JarHell.java (hunk 1 above)
Comment (jasontedor) [inline core/src/main/java/org/elasticsearch/bootstrap/JarHell.java]: I pushed 88e91ae9310182cbc0a1384b517a6d25d9b5cd1b.

**Pull request merged**